plugins/inputs/windows_event_log/wineventlog/wineventlogrecord.go

```diff
@@ -78,3 +82,47 @@ func (record *windowsEventLogRecord) Value() (valueString string, err error) {
 func (record *windowsEventLogRecord) Timestamp() string {
 	return fmt.Sprint(record.System.TimeCreated.SystemTime.UnixNano())
 }
+
+type EventData struct {
+	Values []Data `xml:",any"`
+}
+
+type UserData struct {
+	Values []Data `xml:",any"`
+}
+
+// UserData has slightly different schema than EventData so that we need to overrid this
+// unmarshal function to get similar structure
+//
+// https://learn.microsoft.com/en-us/windows/win32/wes/eventschema-userdatatype-complextype
+// https://learn.microsoft.com/en-us/windows/win32/wes/eventschema-eventdatatype-complextype
+func (u *UserData) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
```

Comment:
Are there existing tests for how we ingest XML windows event log data? Is this covered?

Reply:
We don't have any test cases for the `windowsEventLogRecord` struct, but I agree we should probably not only have test for this `UnmarshalXML` function but also the `windowsEventLogRecord`

Reply:
I've added test cases to cover unmarshaling EventData/UserData. We can add more test cases there to cover the whole windowsEventLogRecord struct in a separate PR.